test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -287,5 +287,38 @@ if (dialect.match(/^postgres/)) {
           });
       });
     });
+
+    describe('addColumn', () => {
+      beforeEach(function() {
+        return this.sequelize.createSchema('archive').then(() => {
+          return this.queryInterface.createTable('users', {
+            id: {
+              type: DataTypes.INTEGER,
+              primaryKey: true,
+              autoIncrement: true
+            }
+          });
+        });
+      });
+
+      it('should work with named enums', function(done) {
+        this.queryInterface.addColumn('users', 'mood', {
+          type: DataTypes.ENUM({
+            name: 'mood_enum',
+            values: ['happy', 'sad', 'neutral']
+          })
+        })
+          .then(() => this.sequelize.query(
+            'SELECT udt_name FROM information_schema.columns WHERE table_name = \'users\' AND column_name = \'mood\'',
+            { type: this.sequelize.QueryTypes.SELECT }
+          ))
+          .then(res => {
+            const actual = res[0].udt_name;
+            expect(actual).to.be.equal('mood_enum');
+            done();
+          })
+          .catch(done);
+      });
```

Comment:
> * In one of the new enum name tests in `.../postgres/query-interface.test.js`, I ended up using the `done()` callback _(which I know is called out as something not to do.)_ , in order to get an accurate failure message.  Otherwise the test just times out and you don't see the error from the failed `expect(...)...`;

Hmmm, I doubt it is really necessary, I suspect you did something wrong, can you please try the code below instead?

```js
it('should work with named enums', function() {
    return this.queryInterface.addColumn('users', 'mood', {
        type: DataTypes.ENUM({
            name: 'mood_enum',
            values: ['happy', 'sad', 'neutral']
        })
    }).then(() => this.sequelize.query(
        'SELECT udt_name FROM information_schema.columns WHERE table_name = \'users\' AND column_name = \'mood\'',
        { type: this.sequelize.QueryTypes.SELECT }
    )).then(res => {
        expect(res[0].udt_name).to.be.equal('mood_enum');
    });
});
```

Reply:
I see, I forgot to return my promise; that works.  Thank you.